tests/framework/test_time.py

```diff
@@ -246,43 +283,106 @@ def test_uneven_steps(SimulationContext, base_config):
     ## 9, etc.
     correct_step_sizes = [3, 3, 1, 2, 3, 2, 1, 3, 3]
     groups = ["evens", "evens", "odds", "evens", "evens", "odds", "evens", "evens", "all"]
-    test = {
-        "evens": sim.get_population().iloc[lambda x: x.index % 2 == 0].index,
-        "odds": sim.get_population().iloc[lambda x: x.index % 2 == 1].index,
-        "all": sim.get_population().index,
-    }
-    pipeline_values = {
-        "evens": pd.Series(from_yearly(1.75, pd.Timedelta(days=3)), index=test["evens"]),
-        "odds": pd.Series(from_yearly(1.75, pd.Timedelta(days=7)), index=test["odds"]),
-        "all": pd.concat(
-            [
-                pd.Series(from_yearly(1.75, pd.Timedelta(days=3)), index=test["evens"]),
-                pd.Series(from_yearly(1.75, pd.Timedelta(days=7)), index=test["odds"]),
-            ]
-        ).sort_index(),
-    }
 
     ## Ensure that steps and active simulants are correct through one cycle of 21
     for correct_step_size, group in zip(correct_step_sizes, groups):
         validate_step_column_is_pipeline(sim)
         take_step_and_validate(
             sim,
             listener,
-            expected_simulants=test[group],
+            expected_simulants=get_pop_by_parity(sim, group).index,
             expected_step_size_days=correct_step_size,
         )
 
         sample_pipeline = step_modifier_component.ts_pipeline_value
-        assert sample_pipeline.index.equals(test[group])
-        assert np.all(sample_pipeline == pipeline_values[group])
+        assert sample_pipeline.index.equals(get_pop_by_parity(sim, group).index)
+        assert np.all(sample_pipeline == pipeline_by_parity(sim, step_modifiers, group))
+
+
+def test_partial_modification(SimulationContext, base_config):
+    """Test that if we have one modifier that doesn't apply to all simulants,
+    we choose the default value for unmodified simulants.
+    """
+
+    base_config["time"]["step_size"] = 1
```

Comment:
In this case we don't set "standard_step_size" so fall back to the "step_size" configuration. Do we have a test where we actually set the "standard_step_size"?